pai/predictor.py

```diff
@@ -296,22 +294,67 @@ def delete_service(self):
         """Delete the service."""
         self.session.service_api.delete(name=self.service_name)
 
-    def wait_for_ready(self):
-        """Wait until the service enter running status."""
+    def wait_for_ready(self, force: bool = False):
+        """Wait until the service enter running status.
+
+        Args:
+            force (bool): Whether to force wait for ready.
+
+        Raises:
+            RuntimeError: Raise if the service terminated unexpectedly.
+
+        """
+        if self.service_status == ServiceStatus.Running and not force:
```

Comment:
wait_for_ready is a public interface that can be used by users. Because the status may not accurately reflect the true state of the service, users can pass force=True to enforce a check of the service status, including the service gateway.